src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -20,10 +21,38 @@ interface TableVisComponentProps {
 }
 
 export const TableVisComponent = ({ table, visConfig, handlers }: TableVisComponentProps) => {
-  const { rows, columns } = table;
+  const { formattedRows: rows, formattedColumns: columns } = convertToFormattedData(
+    table,
+    visConfig
+  );
 
   const pagination = usePagination(visConfig, rows.length);
 
+  // set sort and resize column state
+  const [sortColumn, setSortColumn] = useState<SortColumn>(
+    handlers.uiState.get('vis.sortColumn') || {}
```

Comment:
From what i know about UIState, its a way to get and set state information inside a component rendered by an expression function. We should not be maintaining duplicate state here. If we want to set the new `sortColumn` value, just use the `uiState.set` function. The Visualization should rerender with the updated value automatically.

Reply:
> 

Hey Ashwin, regarding this feedback (https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2502#discussion_r988409245 ), you are right, uiState.set won’t let me to render the component. If I switch setSortColumn(updatedVisState.sortColumn); to uiState.set than it won’t re-render with corrected sorted columns. I am using an example from charts ([https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477f[…]3b/src/plugins/charts/public/static/components/color_schema.tsx](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/charts/public/static/components/color_schema.tsx#L64) ) to force it work. Regarding this part, I could open an issue as a post research topic. Currently, I couldn’t find anything meaningful now.

Reply:
removed local state.